torchvision/ops/ciou_loss.py

```diff
@@ -30,50 +34,25 @@ def complete_box_iou_loss(
             ``'sum'``: The output will be summed. Default: ``'none'``
         eps : (float): small number to prevent division by zero. Default: 1e-7
 
-    Reference:
+    Returns:
+        Tensor: Loss tensor with the reduction option applied.
 
-    Complete Intersection over Union Loss (Zhaohui Zheng et. al)
-    https://arxiv.org/abs/1911.08287
+    Reference:
+        Zhaohui Zheng et. al: Complete Intersection over Union Loss:
+        https://arxiv.org/abs/1911.08287
 
     """
 
-    # Original Implementation : https://github.com/facebookresearch/detectron2/blob/main/detectron2/layers/losses.py
-
     if not torch.jit.is_scripting() and not torch.jit.is_tracing():
         _log_api_usage_once(complete_box_iou_loss)
 
-    boxes1 = _upcast(boxes1)
-    boxes2 = _upcast(boxes2)
+    diou_loss = distance_box_iou_loss(boxes1, boxes2, reduction="none", eps=eps)
+    intsct, union = _loss_inter_union(boxes1, boxes2)
```

Comment:
I would try to minimize speed impact and re-estimation as much as possible. Potentially a private diou method could return intermediate results so that we can reuse intersect and union?

Reply:
Correct, the current refactor re computes stuff and is hence slightly slower. Let me refactor as per your suggestion.